### PR TITLE
Tests when multiple moabs exist for a single druid

### PR DESCRIPTION
**Pull request by jermnelson**
## Why was this change made?
Adds further tests mentioned in ticket #1535 (see https://github.com/sul-dlss/preservation_catalog/issues/1535#issuecomment-644267593).

In the `complete_moab_handler_check_exist_spec.rb`, `complete_moab_handler_confirm_version_spec.rb `, `complete_moab_handler_spec.rb`, and `complete_moab_handler_update_version_spec.rb` tests, an additional `CompleteMoab` is created that shares the same `PreservedObject` in order to test to see if having multiple `CompleteMoab` instances that have the same druid does not violate any of the assumptions that are part of the setup for the tests contained within each of these specs.

## How was this change tested?
Test suite

## Which documentation and/or configurations were updated?
DB Readme updated.

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -74,12 +73,12 @@ Metrics/MethodLength:
...
     - 'app/lib/audit/moab_to_catalog.rb'
     - 'app/lib/preservation_catalog/s3_audit.rb'
-    - 'app/services/audit_results.rb'
     - 'app/services/checksum_validator.rb'
     - 'app/services/complete_moab_handler.rb'
     - 'app/services/moab_storage_root_reporter.rb'
     - 'app/services/moab_validator.rb'
-    - 'app/services/workflow_reporter.rb'
+    - 'app/services/reporters/event_service_reporter.rb'
+    - 'app/services/reporters/workflow_reporter.rb'
```
Comment (aaron-collier) [inline .rubocop_todo.yml:81]: I have a slight preference that this rubocop changes that are affecting master be done in a separate PR (or fixed if possible) as this PR isn't related to the two implicated files. But will defer to @sul-dlss/infrastructure-team thoughts on this.

Comment (jermnelson) [reply .rubocop_todo.yml:81]: Thanks @aaron-collier, not sure the best way to handle when rubocop errors occur on code that the current PR didn't touch but needs to pass in order to be merged. I just ran `rubocop --auto-gen-config --auto-gen-only-exclude` to generate the changes in the *.rubocop_todo.yml* file.

File: db/README.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ _Please keep this up to date as the schema changes!_
   * `druid` is the digital resource unique identifier
   * `current_version` is current latest version we've seen for the druid across all instances.
   * `preservation_policy_id` points to the policy governing how the object should be preserved.
-* A `CompleteMoab` represents a physical copy of a `PreservedObject`, e.g. a moab (represents all versions) stored on premesis and accessed via NFS mount, the `zipped_moab_vesion` (represents a single version) copy sitting on a cloud endpoint (manipulable via REST calls to a 3rd party cloud service), etc.
+* A `CompleteMoab` represents a physical copy of a `PreservedObject`, e.g. a moab (represents all versions) stored on premesis and accessed via NFS mount, the `zipped_moab_vesion` (represents a single version) copy sitting on a cloud endpoint (manipulable via REST calls to a 3rd party cloud service), etc. Multiple `CompleteMoab` instances can share the same `PreservedObject` instance.
```
Comment (mjgiarlo) [inline db/README.md:26]: Are you open to using `belong to` instead of `share`? That makes the relationship a tad clearer and aligns with the language of our tooling (ActiveRecord).

**Comment by jmartin-sul** [comment]
i'm going to go ahead and merge, since i think the feedback left by @aaron-collier and @mjgiarlo has been addressed, and since i don't _think_ the PR has changed significantly since they viewed, other than the fixes related directly to their suggestions.  but aaron and mike should feel free to file follow on tickets if it turns out they had other feedback, or if they noticed things they'd want to change since first reviewing.

but this seems like a definite improvement as-is.  thanks @jermnelson!
